clients/src/main/java/org/apache/kafka/common/network/Selector.java

```diff
@@ -521,8 +523,10 @@ private int select(long ms) throws IOException {
      */
     public void close(String id) {
         KafkaChannel channel = this.channels.get(id);
-        if (channel != null)
+        if (channel != null) {
+            channel.state(ChannelState.LOCAL_CLOSE);
             close(channel, false);
```

Comment:
Yes, that is correct. I have added a comment. When NetworkClient closes connection (e.g. due to a timeout), it invokes `processDisconnection` after invoking `close`, but that doesn't rely on this state.